google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/MessageDispatcher.java

```diff
@@ -105,6 +106,7 @@ class MessageDispatcher {
   private final Distribution ackLatencyDistribution;
 
   private final String subscriptionName;
+  private final SubscriptionName subscriptionNameObject;
```

Comment:
How does this new object interact with the previous `subscriptionName` string? Is the latter still necessary?